src/Schedulers/OrdersScheduler.php

```diff
@@ -0,0 +1,202 @@
+	 *
+	 * @param int      $limit Number of records to retrieve.
+	 * @param int      $page  Page number.
+	 * @param int|bool $days Number of days prior to current date to limit search results.
+	 * @param bool     $skip_existing Skip already imported orders.
+	 */
+	public static function get_items( $limit = 10, $page = 1, $days = false, $skip_existing = false ) {
+		global $wpdb;
+		$where_clause = '';
+		$offset       = $page > 1 ? ( $page - 1 ) * $limit : 0;
+
+		if ( is_int( $days ) ) {
+			$days_ago      = date( 'Y-m-d 00:00:00', time() - ( DAY_IN_SECONDS * $days ) );
+			$where_clause .= " AND post_date >= '{$days_ago}'";
+		}
+
+		if ( $skip_existing ) {
+			$where_clause .= " AND NOT EXISTS (
+				SELECT 1 FROM {$wpdb->prefix}wc_order_stats
+				WHERE {$wpdb->prefix}wc_order_stats.order_id = {$wpdb->posts}.ID
+			)";
+		}
+
+		$count = $wpdb->get_var(
+			"SELECT COUNT(*) FROM {$wpdb->posts}
+			WHERE post_type IN ( 'shop_order', 'shop_order_refund' )
+			AND post_status NOT IN ( 'wc-auto-draft', 'auto-draft', 'trash' )
+			{$where_clause}"
+		); // WPCS: unprepared SQL ok.
+
+		$order_ids = absint( $count ) > 0 ? $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT ID FROM {$wpdb->posts}
+				WHERE post_type IN ( 'shop_order', 'shop_order_refund' )
+				AND post_status NOT IN ( 'wc-auto-draft', 'auto-draft', 'trash' )
+				{$where_clause}
+				ORDER BY post_date ASC
+				LIMIT %d
+				OFFSET %d",
+				$limit,
+				$offset
+			)
+		) : array(); // phpcs:ignore unprepared SQL ok.
+
+		return (object) array(
+			'total' => absint( $count ),
+			'ids'   => $order_ids,
+		);
+	}
+
+	/**
+	 * Get total number of rows imported.
+	 */
+	public static function get_total_imported() {
+		global $wpdb;
+		return $wpdb->get_var( "SELECT COUNT(*) FROM {$wpdb->prefix}wc_order_stats" );
+	}
+
+	/**
+	 * Schedule this import if the post is an order or refund.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public static function possibly_schedule_import( $post_id ) {
+		if ( 'shop_order' !== get_post_type( $post_id ) && 'woocommerce_refund_created' !== current_filter() ) {
+			return;
+		}
+
+		self::schedule_action( 'import', array( $post_id ) );
+	}
+
+	/**
+	 * Imports a single order or refund to update lookup tables for.
+	 * If an error is encountered in one of the updates, a retry action is scheduled.
+	 *
+	 * @param int $order_id Order or refund ID.
+	 * @return void
+	 */
+	public static function import( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		// If the order isn't found for some reason, skip the sync.
+		if ( ! $order ) {
+			return;
+		}
+
+		$type = $order->get_type();
+
+		// If the order isn't the right type, skip sync.
+		if ( 'shop_order' !== $type && 'shop_order_refund' !== $type ) {
+			return;
+		}
+
+		// If the order has no id or date created, skip sync.
+		if ( ! $order->get_id() || ! $order->get_date_created() ) {
+			return;
+		}
+
+		$result = array_sum(
```

Comment:
This block of code is pretty creative :) 

// The update methods return -1 for skip, or a boolean success indicator.

If this is true, the array_sum may be summing booleans. I believe newer versions of PHP throw warnings when types mismatch in math function. You could cast these to `(int)` to prevent that happening.

Or refactor to make clearer, along these lines:

```php
$results = [
     OrdersStatsDataStore::sync_order( $order_id ),
     ProductsDataStore::sync_order_products( $order_id ),
     CouponsDataStore::sync_order_coupons( $order_id ),
     TaxesDataStore::sync_order_taxes( $order_id ),
];

ReportsCache::invalidate();

if ( count( array_intersect( $results, [ -1, false ] ) ) ) {
	self::schedule_action( 'import', array( $order_id ) );
}
```

Reply:
Nice catch, Mike!  All creativity points go to @jeffstieler for working up this solution.

Updated this in https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/e20f159a770a7a28a391e257f91a07291392aee8

Reply:
Nice - checking for `-1` or `false` is much easier to understand and less error prone. 👍